bson/src/main/org/bson/codecs/pojo/PojoBuilderHelper.java

```diff
@@ -158,11 +157,10 @@
                                                                   final Type genericType) {
         PropertyMetadata<T> propertyMetadata = getOrCreatePropertyMetadata(propertyName, declaringClassName, propertyNameMap, typeData);
         if (!isAssignableClass(propertyMetadata.getTypeData().getType(), typeData.getType())) {
-            throw new CodecConfigurationException(format("Property '%s' in %s, has differing data types: %s and %s", propertyName,
+            propertyMetadata.setError(format("Property '%s' in %s, has differing data types: %s and %s.", propertyName,
```

Comment:
Instead of throwing the error during the initial building of the metadata, we save the error, via package private helpers to the property data classes, so it can be inspected during the final validation.